homeassistant/components/updater/__init__.py

```diff
@@ -1,147 +1,243 @@
+            f"{msg['domain']}_{update_details.identifier}_{update_details.available_version}"
+        )
+        updater_data.updates[msg["domain"]].remove(update_details)
+        await updater_data.store.async_save(list(updater_data.skip))
+
+    connection.send_result(
+        msg["id"],
+        _filtered_updates(updater_data),
     )
 
-    # This can take up to 15s which can delay startup
-    asyncio.create_task(coordinator.async_refresh())
 
-    hass.async_create_task(
-        discovery.async_load_platform(hass, Platform.BINARY_SENSOR, DOMAIN, {}, config)
+@websocket_api.websocket_command(
+    {
+        vol.Required("type"): "updater/update",
+        vol.Required("domain"): str,
+        vol.Required("identifier"): str,
+    }
+)
+@websocket_api.async_response
+async def handle_update(
+    hass: HomeAssistant,
+    connection: websocket_api.ActiveConnection,
+    msg: dict,
+):
+    """Handle an update."""
+    updater_data: UpdaterData = hass.data[DOMAIN]
+    update_details = _get_update_details(hass, msg["domain"], msg["identifier"])
+
+    if update_details is None:
+        connection.send_error(
+            msg["id"],
+            "not_found",
+            f"No updates found for {msg['domain']} and {msg['identifier']}",
+        )
+        return
+
+    if not await update_details.update_callback(hass, update_details):
+        connection.send_error(msg["id"], "update_failed", "Update failed")
+        return
+
+    updater_data.updates[msg["domain"]].remove(update_details)
+
+    connection.send_result(
+        msg["id"],
+        _filtered_updates(updater_data),
     )
 
-    return True
-
 
-async def get_newest_version(hass):
-    """Get the newest Home Assistant version."""
-    session = async_get_clientsession(hass)
+@dataclasses.dataclass()
+class UpdaterData:
+    """Data for the updater integration."""
 
-    async with async_timeout.timeout(30):
-        req = await session.get(UPDATER_URL)
+    store: storage.Store
+    skip: set[str]
+    registrations: dict[str, UpdaterRegistration] = dataclasses.field(
+        default_factory=dict
+    )
+    updates: dict[str, list[UpdateDescription]] = dataclasses.field(
+        default_factory=dict
+    )
 
-    try:
-        res = await req.json()
-    except ValueError as err:
-        raise update_coordinator.UpdateFailed(
-            "Received invalid JSON from Home Assistant Update"
-        ) from err
 
-    try:
-        res = RESPONSE_SCHEMA(res)
-        return res["current_version"], res["release_notes"]
-    except vol.Invalid as err:
-        raise update_coordinator.UpdateFailed(
-            f"Got unexpected response: {err}"
-        ) from err
+@dataclasses.dataclass()
+class UpdateDescription:
+    """Describe an updater update."""
+
+    identifier: str
+    name: str
+    current_version: str
+    available_version: str
+    update_callback: Callable[[HomeAssistant, UpdateDescription], Awaitable[bool]]
+    changelog_url: str | None = None
```

Comment:
Should this be a URL? or MD formatted string?

Reply:
Maybe both?
Being able to provide an excerpt as a string (e.g., containing Markdown) and a link to a full log/release details?